File: tabellina.js
// Programma per stampare la tabellina di un numero
// Chiede all'utente il numero della tabellina e fino a quale numero arrivare

// Funzione per chiedere input all'utente
function chiediInput(messaggio) {
    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });
    
    return new Promise((resolve) => {
        rl.question(messaggio, (risposta) => {
            rl.close();
            resolve(parseInt(risposta));
        });
    });
}

// Funzione principale
async function main() {
    try {
        // Chiedi il numero della tabellina
        const numeroTabellina = await chiediInput('Inserisci il numero della tabellina: ');
        
        // Chiedi fino a quale numero arrivare
        const numeroFineTabellina = await chiediInput('Fino a quale numero vuoi arrivare? ');
        
        // Verifica che i numeri siano validi
        if (isNaN(numeroTabellina) || isNaN(numeroFineTabellina)) {
            console.log('Errore: Inserisci numeri validi!');
            return;
        }
        
        if (numeroFineTabellina < 0) {
            console.log('Errore: Il numero finale deve essere positivo!');
            return;
        }
        
        // Stampa la tabellina
        console.log(`\nTabellina del ${numeroTabellina} fino a ${numeroFineTabellina}:`);
        
        const risultati = [];
        for (let i = 0; i <= numeroFineTabellina; i++) {
            risultati.push(numeroTabellina * i);
        }
        
        // Stampa i risultati separati da virgola
        console.log(risultati.join(', '));
        
    } catch (error) {
        console.log('Errore durante l\'esecuzione:', error.message);
    }
}

// Esegui il programma
main();

File: Tabellina.java
import java.util.Scanner;

public class Tabellina {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        try {
            // Chiedi il numero della tabellina
            System.out.print("Inserisci il numero della tabellina: ");
            int numeroTabellina = scanner.nextInt();
            
            // Chiedi fino a quale numero arrivare
            System.out.print("Fino a quale numero vuoi arrivare? ");
            int numeroFineTabellina = scanner.nextInt();
            
            // Verifica che il numero finale sia positivo
            if (numeroFineTabellina < 0) {
                System.out.println("Errore: Il numero finale deve essere positivo!");
                return;
            }
            
            // Stampa la tabellina
            System.out.println("\nTabellina del " + numeroTabellina + " fino a " + numeroFineTabellina + ":");
            
            // Calcola e stampa i risultati
            StringBuilder risultati = new StringBuilder();
            for (int i = 0; i <= numeroFineTabellina; i++) {
                if (i > 0) {
                    risultati.append(", ");
                }
                risultati.append(numeroTabellina * i);
            }
            
            System.out.println(risultati.toString());
            
        } catch (Exception e) {
            System.out.println("Errore: Inserisci numeri validi!");
        } finally {
            scanner.close();
        }
    }
}